subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoReportAggregationPlugin.java

```diff
@@ -0,0 +1,145 @@
+            view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
+            view.lenient(true);
+        });
+
+        Configuration classDirectoriesConf = project.getConfigurations().create("allCodeCoverageReportClassDirectories");
+        classDirectoriesConf.extendsFrom(jacocoAggregation);
+        classDirectoriesConf.setDescription("Supplies the class directories used to produce all aggregated JaCoCo coverage data reports");
+        classDirectoriesConf.setVisible(false);
+        classDirectoriesConf.setCanBeConsumed(false);
+        classDirectoriesConf.setCanBeResolved(true);
+        getJvmPluginServices().configureAsRuntimeClasspath(classDirectoriesConf);
+
+        ArtifactView classDirectories = classDirectoriesConf.getIncoming().artifactView(view -> {
+            view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
+        });
+
+        ReportingExtension reporting = project.getExtensions().getByType(ReportingExtension.class);
+        reporting.getReports().registerBinding(JacocoCoverageReport.class, DefaultJacocoCoverageReport.class);
+
+        // iterate and configure each user-specified report, creating a <reportName>ExecutionData configuration for each
+        reporting.getReports().withType(JacocoCoverageReport.class).configureEach(report -> {
+            report.getReportTask().configure(task -> {
+                // A resolvable configuration to collect JaCoCo coverage data; typically named "testCodeCoverageReportExecutionData"
+                Configuration executionDataConf = project.getConfigurations().create(report.getName() + "ExecutionData");
+                executionDataConf.extendsFrom(jacocoAggregation);
+                executionDataConf.setDescription(String.format("Supplies JaCoCo coverage data to the %s.  External library dependencies may appear as resolution failures, but this is expected behavior.", report.getName()));
+                executionDataConf.setVisible(false);
+                executionDataConf.setCanBeConsumed(false);
+                executionDataConf.setCanBeResolved(true);
+                executionDataConf.attributes(attributes -> {
+                    attributes.attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.class, Category.DOCUMENTATION));
+                    attributes.attribute(DocsType.DOCS_TYPE_ATTRIBUTE, objects.named(DocsType.class, DocsType.JACOCO_COVERAGE));
+                    attributes.attribute(TestType.TEST_TYPE_ATTRIBUTE, report.getTestType().map(tt -> objects.named(TestType.class, tt)));
+                    attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.class, Usage.VERIFICATION));
+                });
+
+                ArtifactView executionData = executionDataConf.getIncoming().artifactView(view -> {
+                    view.componentFilter(id -> id instanceof ProjectComponentIdentifier);
+                    view.lenient(true);
+                });
```

Comment:
To workaround this for now, we can still just wrap the creation of the ArtifactView in a Callable.

Reply:
It might be easier to change the internals of `DefaultConfiguration` so that the logic to compute the attributes is delayed as much as possible.
If `ConfigurationArtifactView` took a `Supplier<ImmutableAttributes>` and invoked that just in time when one of its three methods is called, it should work.